src/control/lib/hardware/pci.go

```diff
@@ -70,25 +111,50 @@ func (pa *PCIAddress) String() string {
 		return ""
 	}
 
-	return fmt.Sprintf("%s:%s:%s.%s", pa.Domain, pa.Bus, pa.Device, pa.Function)
+	fs := pa.FieldStrings()
+	return fmt.Sprintf("%s:%s:%s.%s", fs["Domain"], fs["Bus"], fs["Device"], fs["Function"])
 }
 
-// Equals compares string representation of address.
-func (pa *PCIAddress) Equals(o *PCIAddress) bool {
-	if pa == nil || o == nil {
+// Equals compares two PCIAddress structs for equality.
+func (pa *PCIAddress) Equals(other *PCIAddress) bool {
+	if pa == nil || other == nil {
```

Comment:
If both are nil, shouldn't they be equal?

Reply:
I guess this is a bit of a philosophical question... To me, that would be somewhat surprising behavior. You can't really compare nothing to nothing, and maybe it's splitting hairs, but I think that this kind of comparison function should only operate on the values of a structure, not whether or not it's nil. To put it another way, I think I'd prefer that code which wants to assert that two addresses are nil should do so explicitly.

Reply:
I can roll with that, although it's a little counterintuitive to me. But it's a corner case I don't think we're likely to hit with this structure outside unit tests anyway, so not a huge deal.